### DISCUSSION TITLE
Batch upload markdown files to create wiki web pages

### DISCUSSION THREAD

**Discussion by snoop2head** [requarks/wiki#3386] [Help / Questions]
## Question
* How do I batch upload markdown files to create multiple wiki pages? Markdown files does not contain any images, but are consisted with plain texts only. 
* Any wiki.js API guidelines available for importing markdown files? I've managed to find GraphQL error codes only in [official docs](https://docs.requarks.io/dev/api).

## Host Info (please complete the following information)
* OS: Linux, Docker Container from AWS Marketplace
* Wiki.js version: 2.3.81
* Database engine: PostgreSQL 11.7

**Comment by winkelement** [comment]
The easiest way would be to set up a git repository as a storage strategy then push the .md files to that repo and let Wiki.js do the sync.

See https://docs.requarks.io/storage/git

**Comment by snoop2head** [comment]
@winkelement 
Thanks for the help! Now I imported markdown files in repository to wikipedia. 
* Github repo: https://github.com/snoop2head/wiki-js-storage
* Wiki: http://ec2-3-34-99-110.ap-northeast-2.compute.amazonaws.com/

### Push function problem
But it seems like error to write on github repository occurs! 
> fatal: could not read Username for 'https://github.com': No such device or address

Is the "Default Author Name" section for my id(snoop2head) or my name(Young Jin Ahn) registered on github?

**Comment by winkelement** [comment]
What does your git configuration in Wiki.js looks like?

**Comment by brianturney** [comment]
I want to do this not using GIT. Can someone answer the original question? GraphQL supports file uploads. Please include. 
1) An example of how to list the folderid of the file's upload destination folder. Include examples of how using graphql to do this from the command line in RHEL Linux.
2) An example showing the proper syntax and placement of the API key for use with graphql if applicable. 
2) An example showing how to use graphql to upload a single file. Include examples of how using graphql to do this from the command line in RHEL Linux. Merely showing graphql syntax is insufficient here. I need an example of using graphql from a linux command line. 
3) An example showing how to use graphql to upload multiple files.

**Reply by rrg92** [reply]
You cannot do this. File uploads of graph is just for assets

For import lot of markdown, just with git or using graphsql api to create a page for each file.
